infrastructure/cli/cmd/create/helm.go

```diff
@@ -0,0 +1,218 @@
+			Kind: "ClusterRole",
+			Name: "cluster-admin",
+		},
+	}
+
+	_, err = roleBindingClient.Create(context.TODO(), roleBinding, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (h *Helm) InstallCharts() error {
+	return h.runHelm([]string{"install", "--values", "/apps/config/airy.yaml", "core", "/apps/helm-chart/"})
+}
+
+func (h *Helm) runHelm(args []string) error {
+	if err := h.upsertAiryYaml(); err != nil {
+		return err
+	}
+
+	h.cleanupJob()
+	jobsClient := h.clientset.BatchV1().Jobs(h.namespace)
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      h.name,
+			Namespace: h.namespace,
+			Labels:    map[string]string{"helm-runner": "true"},
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "helm-runner",
+							Image:           "ghcr.io/airyhq/infrastructure/helm:" + h.version,
+							Args:            args,
+							ImagePullPolicy: corev1.PullAlways,
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "core-config",
+									MountPath: "/apps/config",
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "core-config",
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: airyYamlConfigMap,
+									},
+								},
+							},
+						},
+					},
+					RestartPolicy:      "Never",
+					ServiceAccountName: serviceAccountName,
+				},
+			},
+		},
+	}
+
+	_, err := jobsClient.Create(context.TODO(), job, v1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	watcher, err := jobsClient.Watch(context.TODO(), v1.ListOptions{
+		LabelSelector: "helm-runner=true",
+	})
+	if err != nil {
+		return err
+	}
+
+	ch := watcher.ResultChan()
+
+	for event := range ch {
+		switch event.Type {
+		case watch.Error:
+			h.cleanupJob()
+			return fmt.Errorf("helm run failed with error %v", event.Object)
+		case watch.Added:
+		case watch.Modified:
+			job, _ := event.Object.(*batchv1.Job)
+			if job.Status.Succeeded == 1 {
+				h.cleanupJob()
+				return nil
+			} else if job.Status.Failed == 1 {
+				h.cleanupJob()
+				return fmt.Errorf("helm run failed with error %v", event.Object)
+			}
+		}
+	}
+
+	return nil
+}
+
+// Create/update airy yaml config map
+func (h *Helm) upsertAiryYaml() error {
+	cm, _ := h.clientset.CoreV1().ConfigMaps(h.namespace).Get(context.TODO(), airyYamlConfigMap, v1.GetOptions{})
+
+	cmData := map[string]string{
+		"airy.yaml": h.getAiryYaml(),
+	}
+
+	if cm.GetName() == "" {
+		_, err := h.clientset.CoreV1().ConfigMaps(h.namespace).Create(context.TODO(),
+			&corev1.ConfigMap{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      airyYamlConfigMap,
+					Namespace: h.namespace,
+				},
+				Data: cmData,
+			}, v1.CreateOptions{})
+		return err
+	} else {
```

Comment:
we should be able to get by without the else (not the code, just the actual branching)

Reply:
I don't think we can. Creating a config map with the same name will yield an error. I'm probably not getting this correctly